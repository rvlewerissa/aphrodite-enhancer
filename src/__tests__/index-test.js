import styleWithProps from '../index';

global.CURRENTLY_TESTING = true;

const MOCKED_APHRODITE_OBJECT = {
  _len: 209,
  _name: 'headerWrapper_cwpl5d',
  _definition: {
    alignItems: 'center',
    height: '50.6rem',
    backgroundColor: 'rgb(46, 176, 244)'
  }
};

const MOCKED_APHRODITE_OBJECT_2 = {
  _len: 54,
  _name: 'headerLogoText_8bo1c3',
  _definition: {
    color: 'white',
    fontSize: '2.4rem',
    fontWeight: 700
  }
};

describe('StyleWithProps', () => {
  it('should accepts null or undefined', () => {
    expect(styleWithProps(null)).toMatchSnapshot();
  });

  it('should accept plain object', () => {
    expect(styleWithProps({flex: 1, flexDirection: 'row'})).toMatchSnapshot();
  });

  it('should accept array of plain objects', () => {
    expect(
      styleWithProps([
        {flex: 1, flexDirection: 'row'},
        {flex: 2, alignItems: 'center'}
      ])
    ).toMatchSnapshot();
  });

  it('should accept aphrodite object', () => {
    expect(styleWithProps(MOCKED_APHRODITE_OBJECT)).toMatchSnapshot();
  });

  it('should accept array of aphrodite objects', () => {
    expect(
      styleWithProps([MOCKED_APHRODITE_OBJECT, MOCKED_APHRODITE_OBJECT_2])
    ).toMatchSnapshot();
  });

  it('should accept array of mixed objects', () => {
    expect(
      styleWithProps([MOCKED_APHRODITE_OBJECT, {flex: 1, flexDirection: 'row'}])
    ).toMatchSnapshot();
    expect(styleWithProps([null, {flex: 1}])).toMatchSnapshot();
    expect(styleWithProps([MOCKED_APHRODITE_OBJECT, null])).toMatchSnapshot();
  });

  it('should accept nested array and complex mixture of objects', () => {
    expect(
      styleWithProps([
        [MOCKED_APHRODITE_OBJECT, null],
        {flex: 1},
        null,
        MOCKED_APHRODITE_OBJECT_2,
        [[{flexDiretion: 'row'}]]
      ])
    ).toMatchSnapshot();
  });
});
