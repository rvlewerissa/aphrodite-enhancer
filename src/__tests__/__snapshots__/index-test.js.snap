// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StyleWithProps should accept aphrodite object 1`] = `
Object {
  "className": Array [
    Object {
      "_definition": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(46, 176, 244)",
        "height": "50.6rem",
      },
      "_len": 209,
      "_name": "headerWrapper_cwpl5d",
    },
  ],
  "style": Object {},
}
`;

exports[`StyleWithProps should accept array of aphrodite objects 1`] = `
Object {
  "className": Array [
    Object {
      "_definition": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(46, 176, 244)",
        "height": "50.6rem",
      },
      "_len": 209,
      "_name": "headerWrapper_cwpl5d",
    },
    Object {
      "_definition": Object {
        "color": "white",
        "fontSize": "2.4rem",
        "fontWeight": 700,
      },
      "_len": 54,
      "_name": "headerLogoText_8bo1c3",
    },
  ],
  "style": Object {},
}
`;

exports[`StyleWithProps should accept array of mixed objects 1`] = `
Object {
  "className": Array [
    Object {
      "_definition": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(46, 176, 244)",
        "height": "50.6rem",
      },
      "_len": 209,
      "_name": "headerWrapper_cwpl5d",
    },
  ],
  "style": Object {
    "flex": 1,
    "flexDirection": "row",
  },
}
`;

exports[`StyleWithProps should accept array of mixed objects 2`] = `
Object {
  "className": Array [],
  "style": Object {
    "flex": 1,
  },
}
`;

exports[`StyleWithProps should accept array of mixed objects 3`] = `
Object {
  "className": Array [
    Object {
      "_definition": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(46, 176, 244)",
        "height": "50.6rem",
      },
      "_len": 209,
      "_name": "headerWrapper_cwpl5d",
    },
  ],
  "style": Object {},
}
`;

exports[`StyleWithProps should accept array of plain objects 1`] = `
Object {
  "className": Array [],
  "style": Object {
    "alignItems": "center",
    "flex": 2,
    "flexDirection": "row",
  },
}
`;

exports[`StyleWithProps should accept nested array and complex mixture of objects 1`] = `
Object {
  "className": Array [
    Object {
      "_definition": Object {
        "alignItems": "center",
        "backgroundColor": "rgb(46, 176, 244)",
        "height": "50.6rem",
      },
      "_len": 209,
      "_name": "headerWrapper_cwpl5d",
    },
    Object {
      "_definition": Object {
        "color": "white",
        "fontSize": "2.4rem",
        "fontWeight": 700,
      },
      "_len": 54,
      "_name": "headerLogoText_8bo1c3",
    },
  ],
  "style": Object {
    "flex": 1,
    "flexDiretion": "row",
  },
}
`;

exports[`StyleWithProps should accept plain object 1`] = `
Object {
  "className": Array [],
  "style": Object {
    "flex": 1,
    "flexDirection": "row",
  },
}
`;

exports[`StyleWithProps should accepts null or undefined 1`] = `
Object {
  "className": Array [],
  "style": Object {},
}
`;
